### PR TITLE
Add --no-timeout

### PR DESCRIPTION
**Pull request by Papierkorb**
Hello!

I recently found this tool to hide the mouse cursor if I'm using my notebooks touch screen.  And it works great!

However, my small gripe is that I need to set a timeout for the cursor to be hidden anyway.   Thus, I've added the `--no-timeout` option which allows the user to disable the timeout functionality entirely.

The defaults are unchanged, so existing users shouldn't notice much difference to their work flows :)

Cheers!

### CODE REVIEW

File: src/event.c [hunk 1]
```diff
@@ -22,7 +22,7 @@ void event_init(void) {
 
     loop = EV_DEFAULT;
     event_init_x_loop();
-    event_init_timer();
+    if (config.timeout >= 0.0) event_init_timer();
```
Comment (Airblader) [inline src/event.c:25]: nit: could we please put this on a separate line and use braces?

File: src/event.c [hunk 2]
```diff
@@ -102,7 +102,7 @@ static void x_check_cb(EV_P_ ev_check *w, int revents) {
             /* We don't bother checking the exact event since we only select events that interest us. */
             cursor_show();
         }
-        ev_timer_again(loop, idle_watcher);
+        if (idle_watcher) ev_timer_again(loop, idle_watcher);
```
Comment (Airblader) [inline src/event.c:105]: nit: could we please put this on a separate line and use braces?

**Review by Airblader — CHANGES_REQUESTED** [review]
Thanks for the PR! I just have two minor style nitpicks. Could you please take care of them & ideally squash the commits? Thanks so much!

Please do drop a comment when you're done so I get notified.

**Comment by Papierkorb** [comment]
Pushed with the requested changes

**Review by Airblader — APPROVED** [review]
Awesome, thanks!

**Pull request merged**